Composer/packages/extensions/adaptive-form/src/utils/getUIOptions.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { UISchema, UIOptions, JSONSchema7 } from '@bfc/extension';
+/**
+ * Merges overrides and plugins into default ui schema and returns the UIOptions
+ */
+export function getUIOptions(schema?: JSONSchema7, uiSchema?: UISchema, pluginOptions?: UIOptions): UIOptions {
+  const { $kind } = schema?.properties || {};
+
+  const kind = $kind && typeof $kind === 'object' && ($kind.const as string);
+
+  const formOptions = uiSchema && kind && uiSchema[kind] ? uiSchema[kind] : {};
+
+  return { ...formOptions, ...pluginOptions };
```

Comment:
Let's invert this so that formOptions override pluginOptions.

Reply:
Will do, although intuitively it feels to me like a plugin should be able to overwrite defaults.

Reply:
These are the merged result of all the plugins. Which have overwritten the defaults already.

Just in this case, I think that specifying something on a particular field should take precedence over an override for a role.